### PR TITLE
Update README.md to include clear authentication/access control and steps to secure deployed app

### PR DESCRIPTION
**Pull request by andyloree**
Added verbiage for enabling authentication and access controls to restrict access.

## Purpose
Currently, the azd deployment enables no authentication or access control to the web application, potentially exposing customers using this demo application against their own data to unrestricted access, even when done within restricted vnet/private endpoint environments.  This PR requests a section specifically be added to the README to state this clearly to users, and give them links to how to enable authentication and access control.
* ...

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  N/A documentation only update, review changes to README.md

## What to Check
*  N/A documentation only update, review changes to README.md

## Other Information
*  None

### CODE REVIEW

**Comment by pamelafox** [comment]
Thanks @andyloree ! In my AAD PR, I added an FAQ about this, but I commented there that I wasn't sure if it should be a top-level section. It might be important enough to call out from a security perspective. If we keep this as a top level PR, then I can add to it in the AAD PR instead of having an FAQ.

File: README.md [hunk 1]
```diff
@@ -134,6 +134,9 @@ Once in the web app:
 * Explore citations and sources
 * Click on "settings" to try different options, tweak prompts, etc.
 
+### Enable authorization/access control and restricted access
```
Comment (pamelafox) [inline README.md:137]: I'd suggest titling it shorter for consistency

```suggestion
### Enabling authentication
```

File: README.md [hunk 2]
```diff
@@ -134,6 +134,9 @@ Once in the web app:
...
+By default, the Azure web app deployed will have no authentication or access restrictions enabled, meaning anyone with routable network access to the web app can access and search.  You can require authentication to your Azure Active Directory by following the [Add app authentication](https://learn.microsoft.com/en-us/azure/app-service/scenario-secure-app-authentication-app-service) tutorial and set it up against the backend web app deployed earlier.  To then limit access to a specific set of users or groups, you can then follow the steps from [Restrict your Azure AD app to a set of users](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-restrict-your-app-to-a-set-of-users) by changing ''Assignment Required?'' option under the Enterprise Application, and then assigning users/groups access.  Users not granted explicit access will recieve the error message -AADSTS50105: Your administrator has configured the application <app_name> to block users unless they are specifically granted ('assigned') access to the application.-
```
Comment (pamelafox) [inline README.md:138]:
```suggestion
By default, the deployed Azure web app will have no authentication or access restrictions enabled, meaning anyone with routable network access to the web app can chat with your indexed data.  You can require authentication to your Azure Active Directory by following the [Add app authentication](https://learn.microsoft.com/azure/app-service/scenario-secure-app-authentication-app-service) tutorial and set it up against the deployed web app.

To then limit access to a specific set of users or groups, you can follow the steps from [Restrict your Azure AD app to a set of users](https://learn.microsoft.com/azure/active-directory/develop/howto-restrict-your-app-to-a-set-of-users) by changing "Assignment Required?" option under the Enterprise Application, and then assigning users/groups access.  Users not granted explicit access will receive the error message -AADSTS50105: Your administrator has configured the application <app_name> to block users unless they are specifically granted ('assigned') access to the application.-
```

**Comment by andyloree** [comment]
> Thanks @andyloree ! In my AAD PR, I added an FAQ about this, but I commented there that I wasn't sure if it should be a top-level section. It might be important enough to call out from a security perspective. If we keep this as a top level PR, then I can add to it in the AAD PR instead of having an FAQ.

Sure thing @pamelafox .  I am good with whatever organization/changes, I would love at least a mention of no auth in the README (maybe in the FAQ?), just so customers deploying this have a clear understanding of the risks, since most are pointing at their own data.

**Review by pamelafox — APPROVED** [review]

**Pull request merged**